### ISSUE TITLE
Is NetFramework 4.7.2 runtime supported?

### ISSUE THREAD

**Issue by valorekhov** [fsharplang-ru/pulsar-client-dotnet#89]
I am encountering an issue where running a basic code snippet to connect a producer client runs into ConnectException's while executing on the .NET Framework 4.7 runtime. The same code works without any issues on the NetCoreApp 3.1 runtime. While on .NET Framework, the server shows multiple repeated connectivity attempts with incremental backoff spacing until  the .NET Framework -based producer client  throws the `Pulsar.Client.Common.ConnectException` after about 30 seconds on the last line of the included sample below:

```c#
           _client = new PulsarClientBuilder()
                .ServiceUrl("pulsar://server.fqdn:6650")
                .Build();
 
            _clientContactProducer = await new ProducerBuilder(_client).Topic("client-contact")
                .CreateAsync();
```

Will be grateful for any pointers to fix/debug this issue on  .NET Framework.

Thanks!
Val

**Comment by Lanayx** [comment]
It is supported, usually the issue lies in missing/incorrect Nuget dependencies, try running the example using VS debug and check ongoing exceptions in the Events tab. This helps when exception is not visible in logs. Meanwhile I'm going to add an example project on Full .NET with all required config tweaks.

**Comment by Lanayx** [comment]
Added an example for the oldest framework we support - dotnet461, you can check it up [here](https://github.com/fsharplang-ru/pulsar-client-dotnet/tree/develop/examples/OldDotnetExample)

**Comment by valorekhov** [comment]
Thanks so much for putting the NetFramework sample together. It did work in my environment. 
I will be closing this issue soon once I figure out the delta between the sample and my project setup

**Comment by valorekhov** [comment]
I was successfully able to upgrade dependencies in my project using info on versions from the sample project. 

Although I initially attempted to identify the specific faulty dependency by attempting to upgrade deps peacemeal, I wasn't able to finish this task. 

I ended up instead pasting over the contents of packages.config with the contents from the sample project. Since HintPath references in the .csproj file continued to point to the wrong dep versions, I also ended up converting from package.config to the [PackageReference format](https://docs.microsoft.com/en-us/nuget/consume-packages/migrate-packages-config-to-package-reference), which positively solved this issue.

Thank you for your help and your work in this project!
